File: translations/codenet/deepseek_commented/Go2C++/deepseek/Code/s783520158.cpp
#include <iostream>
using namespace std;

int main() {
    // Declare three integer variables to hold the input values
    int a, b, c;
    
    // Read three integer values from the standard input and assign them to a, b, and c
    cin >> a >> b >> c;
    
    // Check if the difference between b and a is equal to the difference between c and b
    if (b - a == c - b) {
        // If the condition is true, print "YES"
        cout << "YES" << endl;
    } else {
        // Otherwise, print "NO"
        cout << "NO" << endl;
    }
    
    return 0;
}

File: translations/codenet/mistral_commenetd/Go2J/starcoder/Code/s783520158.java
package main;

import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		// Declare three integer variables "a", "b", and "c"
		int a, b, c;

		// Read three integers from the standard input and assign them to the variables
		Scanner sc = new Scanner(System.in);
		a = sc.nextInt();
		b = sc.nextInt();
		c = sc.nextInt();

		// Check if the difference between "b" and "a" is equal to the difference between "c" and "b"
		if (b - a == c - b) {
			// If the condition is true, print "YES" to the standard output
			System.out.println("YES");
		} else {
			// If the condition is false, print "NO" to the standard output
			System.out.println("NO");
		}

		// End of the code
	}

}

